clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestState.java

```diff
@@ -66,13 +65,19 @@ public boolean canSendRequest(final long currentTimeMs) {
             return true;
         }
 
-        if (this.lastReceivedMs == -1 ||
-                this.lastReceivedMs < this.lastSentMs) {
-            // there is an inflight request
+        if (this.lastReceivedMs == -1 || this.lastReceivedMs < this.lastSentMs) {
+            log.debug("An inflight request already exists; lastReceivedMs: {}, lastSentMs:  {}", lastReceivedMs, lastSentMs);
             return false;
         }
 
-        return requestBackoffExpired(currentTimeMs);
+        long remainingBackoffMs = remainingBackoffMs(currentTimeMs);
+
+        if (remainingBackoffMs <= 0) {
+            return true;
+        } else {
```

Comment:
just a note: A lot of code in AK seems like to explicitly using else despite already returned early.  Technically, we don't need the else.